ui/src/router/components/form/components/ensembler_config/typeOptions.js

```diff
@@ -1,68 +1,101 @@
-      </Fragment>
-    ),
-  },
-  {
-    value: "external",
-    inputDisplay: "External (Coming Soon)",
-    description: (
-      <Fragment>
-        Turing will send responses from all routes, together with treatment
-        configuration, to the external URL for ensembling
-      </Fragment>
-    ),
-    disabled: true,
-  },
-];
+const typeOptions = {
+  "HTTP_JSON": [
+    {
+      value: "nop",
+      inputDisplay: "None",
+      description: (
+        <Fragment>
+          Turing will simply return response from the route marked as the{" "}
+          <strong>Final Response</strong>
+        </Fragment>
+      ),
+    },
+    {
+      value: "standard",
+      inputDisplay: "Standard",
+      description: (
+        <Fragment>
+          Turing will select the response from one of the routes, based on the
+          configured mapping between routes and experiment treatments
+        </Fragment>
+      ),
+    },
+    {
+      value: "docker",
+      inputDisplay: "Docker",
+      description: (
+        <Fragment>
+          Turing will deploy specified image as a post-processor and will send to
+          it responses from all routes, together with the treatment configuration,
+          for the ensembling
+        </Fragment>
+      ),
+    },
+    {
+      value: "pyfunc",
+      inputDisplay: "Pyfunc",
+      description: (
+        <Fragment>
+          Turing will build and deploy the selected pyfunc ensembler and will send
+          to it responses from all routes, together with the treatment
+          configuration, for ensembling
+        </Fragment>
+      ),
+    },
+    {
+      value: "external",
+      inputDisplay: "External (Coming Soon)",
+      description: (
+        <Fragment>
+          Turing will send responses from all routes, together with treatment
+          configuration, to the external URL for ensembling
+        </Fragment>
+      ),
+      disabled: true,
+    },
+  ],
+  "UPI_V1": [
+    {
+      value: "nop",
+      inputDisplay: "None",
+      description: (
+        <Fragment>
+          Turing will simply return response from the route marked as the{" "}
+          <strong>Final Response</strong>. Other ensembler type not support for UPI yet.
+        </Fragment>
+      ),
+    },
+    {
+      value: "standard",
+      inputDisplay: "Standard",
+      description: (
+        <Fragment>
+          Turing will select the response from one of the routes, based on the
+          configured mapping between routes and experiment treatments. Only standard 
+          ensembler is supported for UPI now.
+        </Fragment>
+      ),
+    }
+  ]
+};
 
-export const ensemblerTypeOptions = (engineProps) => {
+export const ensemblerTypeOptions = (engineProps, protocol) => {
+  const options = typeOptions[protocol]
+  // For UPI, only standard ensembler is available and must be selected when there is experiment engine
+  if (protocol === "UPI_V1") {
+    if (engineProps.name) {
```

Comment:
`if (engineProps.name)` -> `if (!!engineProps.name)` is more apt.

In any case, why are we doing a special check with the protocol here? Doesn't `const options = typeOptions[protocol]` already ensure we have the right subset? The check that comes after should be enough, right?

```javascript
if (!engineProps.name) {
    // Standard Ensembler is not available when there is no experiment engine
    return options.filter((o) => o.value !== "standard");
}
```

Reply:
right, i can remove it totally. I added this before I added the map and missed updating this 🙏